source/Calamari/Deployment/PackageRetention/Repositories/JournalJsonRepositoryFactory.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Calamari.Common.Features.Processes.Semaphores;
+using Calamari.Common.Plumbing.FileSystem;
+
+namespace Calamari.Deployment.PackageRetention.Repositories
+{
+    public class JournalJsonRepositoryFactory : IJournalRepositoryFactory
+    {
+        readonly ICalamariFileSystem fileSystem;
+        readonly ISemaphoreFactory semaphoreFactory;
+        readonly string journalPath;
+
+        public JournalJsonRepositoryFactory(ICalamariFileSystem fileSystem, ISemaphoreFactory semaphoreFactory)//, string journalPath)
+        {
+            this.fileSystem = fileSystem;
+            this.semaphoreFactory = semaphoreFactory;
+            this.journalPath = "c:\\Octopus\\journal.json"; 
```

Comment:
This should come from the server, I think?

Reply:
I think same mechanism as the deployment journal. Currently is set via env variable coming from Tentacle